packages/cloudflare/test/handler.test.ts

```diff
@@ -0,0 +1,300 @@
+// Although this is not ideal, this is the best we can do until we have a better way to test cloudflare workers.
+
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+
+import * as SentryCore from '@sentry/core';
+import type { Event } from '@sentry/types';
+import { CloudflareClient } from '../src/client';
+import { withSentry } from '../src/handler';
+
+const MOCK_ENV = {
+  SENTRY_DSN: 'https://public@dsn.ingest.sentry.io/1337',
+};
+
+describe('withSentry', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  test('gets env from handler', async () => {
+    const handler = {
+      fetch(_request, _env, _context) {
+        return new Response('test');
+      },
+    } satisfies ExportedHandler;
+
+    const optionsCallback = vi.fn().mockReturnValue({});
+
+    const wrappedHandler = withSentry(optionsCallback, handler);
+    await wrappedHandler.fetch(new Request('https://example.com'), MOCK_ENV, createMockExecutionContext());
+
+    expect(optionsCallback).toHaveBeenCalledTimes(1);
+    expect(optionsCallback).toHaveBeenLastCalledWith(MOCK_ENV);
+  });
+
+  test('passes through the response from the handler', async () => {
+    const response = new Response('test');
+    const handler = {
+      async fetch(_request, _env, _context) {
+        return response;
+      },
+    } satisfies ExportedHandler;
+
+    const wrappedHandler = withSentry(() => ({}), handler);
+    const result = await wrappedHandler.fetch(
+      new Request('https://example.com'),
+      MOCK_ENV,
+      createMockExecutionContext(),
+    );
+
+    expect(result).toBe(response);
+  });
+
+  test('flushes the event after the handler is done using the cloudflare context.waitUntil', async () => {
+    const handler = {
+      async fetch(_request, _env, _context) {
+        return new Response('test');
+      },
+    } satisfies ExportedHandler;
+
+    const context = createMockExecutionContext();
+    const wrappedHandler = withSentry(() => ({}), handler);
+    await wrappedHandler.fetch(new Request('https://example.com'), MOCK_ENV, context);
+
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(context.waitUntil).toHaveBeenCalledTimes(1);
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(context.waitUntil).toHaveBeenLastCalledWith(expect.any(Promise));
+  });
+
+  test('creates a cloudflare client and sets it on the handler', async () => {
+    const handler = {
+      async fetch(_request, _env, _context) {
+        expect(SentryCore.getClient() instanceof CloudflareClient).toBe(true);
+        return new Response('test');
+      },
+    } satisfies ExportedHandler;
+
+    const context = createMockExecutionContext();
+    const wrappedHandler = withSentry(() => ({}), handler);
+    await wrappedHandler.fetch(new Request('https://example.com'), MOCK_ENV, context);
+
+    expect.assertions(1);
```

Comment:
m: I think a robuster way to write this is to create a `vi.fn` that gets called with `SentryCore.getClient()` and assert that it has been called with that.

Reply:
Good call!

Changed with 61ad09f395796a814ea8ceefd3e5f333f41dad34